cmake/CmakeFunctionHelper.cmake

```diff
@@ -26,18 +26,27 @@
 set(executor_list "PolicyHandler<codeplay_policy>")
 #represent the list of supported index/increment type
 set(index_list "int" )
-#represent the list of supported data type.
+
+# BLAS_DATA_TYPES represents the list of supported data type.
 #Each data type in a data list determines the container types.
 #The container type for SYCLbackend is BufferIterator<${data}, codeplay_policy>
-set(data_list "float")
- #if double supported we add double as a data type
-if(DOUBLE_SUPPORT)
-  set(data_list "float" "double")
-endif()
 
 ## represent the list of bolean options
 set(boolean_list "true" "false")
 
+# Maps a user provided data type name to the C++ type
+# See BLAS_DATA_TYPES
+function(to_cpp_type output data)
+  set(${output} "${data}" PARENT_SCOPE)
+endfunction()
+
+# Strips some C++ symbols from the string to make it suitable to be used
+# as part of a filename
+function(cpp_type_to_name output cpp_type)
```

Comment:
You might be interested in https://cmake.org/cmake/help/latest/command/string.html#make-c-identifier

Reply:
Not quite what I wanted because it leads to longer names, but I agree it's better to use a standard function: https://github.com/codeplaysoftware/sycl-blas/pull/245/commits/da4befce878c6de889783bfb5e38056d8977289e